### PR TITLE
Refactor cli v.0.6.1

### PR DESCRIPTION
**Pull request by jtenner**
This pull request refactors the functions out into their own separate modules.

Later changes to functionality will be deferred for v 0.7.0

### CODE REVIEW

File: src/cli/cli.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
+
```
Comment (willemneal) [inline src/cli/cli.ts]: My one complaint is why is this not `index.ts`?  Then you wouldn't have to change the import path other places.

File: src/cli/run.ts [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+    "--debug": [],
+    "--measure": [],
+    "--sourceMap":[],
+    /** This is required. Do not change this. */
+    "--binaryFile": ["output.wasm"],
+  };
+  const disclude: RegExp[] = configuration.disclude || [];
+  let reporter: TestReporter = configuration.reporter || new DefaultTestReporter();
+
+  const performanceConfiguration = configuration.performance || createDefaultPerformanceConfiguration();
```
Comment (willemneal) [inline src/cli/run.ts]: One other idea is to break this out as it's own class which you pass yargs in the constructor.  Then the getters would handle the default case or read from yargs.

File: src/cli/cli.ts (hunk 1 above)
Comment (jtenner) [reply src/cli/cli.ts]: This sounds like good criticism. I'll make the change.

File: src/cli/run.ts (hunk 2 above)
Comment (jtenner) [reply src/cli/run.ts]: I don't understand what you mean?

Comment (willemneal) [reply src/cli/run.ts]:
```ts
class PerformanceConfiguration {
    constructor(private args: yargs){}

    get maxSamples(): string {
        return this.args.argv.maxSamples ? parseFloat(this.args.argv.maxSamples.toString()) : null;
    }
}
```

Comment (jtenner) [reply src/cli/run.ts]: I will refactor this section in the next pull request. It's a breaking change and should get a major version bump.

**Pull request merged**